index.html

```diff
@@ -0,0 +1,235 @@
+          VideoChat.peerConnection.addIceCandidate(rtcCandidate);
+        },
+      
+        // Create an offer that contains the media capabilities of the browser.
+        createOffer: function(){
+          VideoChat.peerConnection.createOffer(
+            function(offer){
+              // If the offer is created successfully, set it as the local description
+              // and send it over the socket connection to initiate the peerConnection
+              // on the other side.
+              VideoChat.peerConnection.setLocalDescription(offer);
+              VideoChat.socket.emit('offer', JSON.stringify(offer));
+            },
+            function(err){
+              // Handle a failed offer creation.
+              console.log(err);
+            }
+          );
+        },
+      
+        // Create an answer with the media capabilities that both browsers share.
+        // This function is called with the offer from the originating browser, which
+        // needs to be parsed into an RTCSessionDescription and added as the remote
+        // description to the peerConnection object. Then the answer is created in the
+        // same manner as the offer and sent over the socket.
+        createAnswer: function(offer){
+          console.log('\nOnMedia\n');
+          return function(){
+            var rtcOffer = new RTCSessionDescription(JSON.parse(offer));
+            VideoChat.peerConnection.setRemoteDescription(rtcOffer);
+            VideoChat.peerConnection.createAnswer(
+              function(answer){
+                VideoChat.peerConnection.setLocalDescription(answer);
+                VideoChat.socket.emit('answer', JSON.stringify(answer));
+              },
+              function(err){
+                // Handle a failed answer creation.
+                console.log(err);
+              }
+            );
+          }
+        },
+      
+        // When a browser receives an offer, set up a callback to be run when the
+        // ephemeral token is returned. 
+        onOffer: function(offer){
+          console.log('\n Video offer');
+          VideoChat.socket.on('token', VideoChat.onToken(VideoChat.createAnswer(offer)));
+          VideoChat.socket.emit('token');
+        },
+      
+        // When an answer is received, add it to the peerConnection as the remote
+        // description.
+        onAnswer: function(answer){
+          var rtcAnswer = new RTCSessionDescription(JSON.parse(answer));
+          VideoChat.peerConnection.setRemoteDescription(rtcAnswer);
+        },
+      
+        // When the peerConnection receives the actual media stream from the other
+        // browser, add it to the other video element on the page.
+        onAddStream: function(event){
+          VideoChat.remoteVideo = document.getElementById('remote-video');
+          VideoChat.remoteVideo.srcObject = event.stream;
+        }
+      };
+      
+      // Get the video button and add a click listener to start the getUserMedia
```

Comment:
idk if this is some JS format thing, but why are we limiting comment line length to 80 characters? IMO having 100 or 120 is clearer